pkg/scalers/pulsar_scaler.go

```diff
@@ -0,0 +1,301 @@
+	stats := new(pulsarStats)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", s.metadata.statsURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+
+	res, err := s.client.Do(req)
+	if res == nil || err != nil {
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+
+	defer res.Body.Close()
+
+	switch res.StatusCode {
+	case 200:
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return nil, fmt.Errorf("error requesting stats from url: %s", err)
+		}
+		err = json.Unmarshal(body, stats)
+		if err != nil {
+			return nil, fmt.Errorf("error unmarshalling response: %s", err)
+		}
+		return stats, nil
+	case 404:
+		return nil, nil
+	default:
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+}
+
+func (s *pulsarScaler) getMsgBackLog(ctx context.Context) (int64, bool, error) {
+	stats, err := s.GetStats(ctx)
+	if err != nil {
+		return 0, false, err
+	}
+
+	if stats == nil {
+		return 0, false, nil
+	}
+
+	v, found := stats.Subscriptions[s.metadata.subscription]
+
+	return v.Msgbacklog, found, nil
+}
+
+// IsActive determines if we need to scale from zero
+func (s *pulsarScaler) IsActive(ctx context.Context) (bool, error) {
+	msgBackLog, found, err := s.getMsgBackLog(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	if !found || msgBackLog <= s.metadata.activationTargetQueryValue {
+		pulsarLog.Info("Pulsar subscription is not active, either no subscription found or no backlog detected")
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// GetMetrics returns value for a supported metric and an error if there is a problem getting the metric
+func (s *pulsarScaler) GetMetrics(ctx context.Context, metricName string, _ labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	msgBacklog, found, err := s.getMsgBackLog(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+
+	if !found {
+		return nil, nil
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *resource.NewQuantity(msgBacklog, resource.DecimalSI),
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+func (s *pulsarScaler) GetMetricSpecForScaling(context.Context) []v2beta2.MetricSpec {
+	targetMetricValue := resource.NewQuantity(s.metadata.msgBacklogThreshold, resource.DecimalSI)
+
+	metricName := fmt.Sprintf("%s-%s-%s", "pulsar", s.metadata.topic, s.metadata.subscription)
+
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: GenerateMetricNameWithIndex(s.metadata.scalerIndex, kedautil.NormalizeString(metricName)),
+		},
+		Target: v2beta2.MetricTarget{
+			Type:         v2beta2.AverageValueMetricType,
+			AverageValue: targetMetricValue,
+		},
+	}
+	metricSpec := v2beta2.MetricSpec{External: externalMetric, Type: pulsarMetricType}
+	return []v2beta2.MetricSpec{metricSpec}
+}
```

Comment:
Some considerations here:
- Use the function `GetMetricTarget` to generate the target section (a sample [here](https://github.com/kedacore/keda/blob/main/pkg/scalers/artemis_scaler.go#L270-L279))
- I'd move the `metricName` generation to the parsing section to avoid parsing it every time